web_app/views/auth.py

```diff
@@ -1,35 +1,118 @@
         form = SignUpForm(request.POST)
         if form.is_valid():
             form.save()
-            username = form.cleaned_data.get('username')
-            raw_password = form.cleaned_data.get('password1')
+            username = form.cleaned_data.get("username")
+            raw_password = form.cleaned_data.get("password1")
             user = authenticate(username=username, password=raw_password)
             login(request, user)
-            return redirect('index')
+            return redirect("web_app:index")
     else:
         form = SignUpForm()
-    return render(request, 'web_app/auth/register.html', {'form': form})
+    return render(request, "web_app/auth/register.html", {"form": form})
+
 
 def login_view(request):
-    if request.method == 'POST':
-        username = request.POST['username']
-        password = request.POST['password']
-        user = authenticate(request, username=username, password=password)
-        if user is not None:
-            login(request, user)
-            return redirect('index')
-    form = AuthenticationForm()
-    return render(request, 'web_app/auth/login.html', {'form': form})
+    if request.method == "POST":
+        form = AuthenticationForm(data=request.POST)
+        print("form is valid: ", form.is_valid())
+        if form.is_valid():
+            username = request.POST["username"]
+            password = request.POST["password"]
+            user = authenticate(request, username=username, password=password)
+            if user is not None and user.is_active:
+                login(request, user)
+                return redirect("web_app:index")
+    else:
+        form = AuthenticationForm()
+    return render(request, "web_app/auth/login.html", {"form": form})
+
+
+def activation_view(request, code: str):
+    if request.method == "POST":
+        form = ActivationForm(user=None, data=request.POST)
+        activation_record = get_object_or_404(Activation, code=code)
+        form.user = activation_record.user
+        code_is_valid(code, request.POST["email"], activation_record.user)
+        if form.is_valid():
+            try:
+                signer = TimestampSigner()
+                plain_text_email = signer.unsign(
+                    request.POST.get("email") + ":" + activation_record.code
+                )
+                if (
+                    plain_text_email == request.POST.get("email")
+                    and plain_text_email == form.user.email
+                ):
+                    form.user.is_active = True
+                    form.save()  # save the activation form (password set)
+                    form.user.save()  # save the users is_active flag
+
+                    user = authenticate(
+                        request,
+                        username=form.user.username,
+                        password=form.data["new_password1"],
+                    )
+                    if user is not None and user.is_active:
```

Comment:
Is there a false-y value of user which needs to be accommodated here?  If not, writing this as `if user ...` (vs. `if user  is not None ...`) should be sufficient.

Reply:
Agreed, I will update this to `if not user`.